### PR TITLE
WIP: Create <ValidatedInput> component.

### PR DESCRIPTION
**Pull request by awong-dev**

### CODE REVIEW

File: _health-care/_js/_components/validated-input.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * `required` - boolean. When present, the field may not be blank.
+ * `value` - string. Value of the input field.
+ * `onChange` - a function with this prototype: (newValue, isValid)
+ */
+class ValidatedInput extends React.Component {
+  propTypes: {
+    errorMessage: React.PropTypes.string,
+//    pattern: React.PropTypes.instanceOf(RegExp),
+    placeholder: React.PropTypes.string,
+    required: React.PropTypes.bool,
```
Comment (akainic) [inline _health-care/_js/_components/validated-input.jsx]: I'm assuming not all the propTypes are included yet?

File: _health-care/_js/_components/validated-input.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  handleInputChange() {
+    const fieldValue = this.refs.textInput.value;
+
+    if (!this.validate(fieldValue)) {
+      this.setState({ hasError: true });
+    } else {
+      this.setState({ hasError: false });
+    }
+
+    this.props.onChange(this.state.hasError, fieldValue);
```
Comment (akainic) [inline _health-care/_js/_components/validated-input.jsx]: What's the vision for the prop `onChange` on `<ValidatedInput />`? Is this going to replace those 2 separate `passUp` functions I had and passing both things that need to be updated in one function?

File: _health-care/_js/_components/validated-input.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import _ from 'lodash';
+
+import ErrorMessage from './error-message';
+
+/**
+ * An text input field that can regexp validate an input.
+ *
+ * Validation has the following props.
+ * `pattern` - a regexp that will be used to validate the input.
```
Comment (akainic) [inline _health-care/_js/_components/validated-input.jsx]: Oh that's a good idea! That way we don't need a ton of individual validation functions. Nice :)

**Comment by akainic** [comment]
How is `<ValidatedInput />` going to be used? Does it replace the `<TextInput />` component?

File: _health-care/_js/_components/social-security-number.jsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ class SocialSecurityNumber extends React.Component {
       <div>
         <div className={`usa-input-grid usa-input-grid-medium ${errorClass}`}>
           <label className={`${errorClass}-label`} htmlFor={`${this.id}_snn`}>Social Security Number
-            <span className="usa-additional_text">Required</span>
+            <span className="usa-additional_text">Required Hi</span>
```
Comment (akainic) [inline _health-care/_js/_components/social-security-number.jsx]: Was this just for testing purposes? Can we remove it now?

**Comment by awong-dev** [comment]
Oops. I should have just closed this PR. It's obsoleted by #1246 . Closing now.

**Pull request closed without merging**